workspace/blink/__fixtures__/arduino.cpp

```diff
@@ -1615,6 +1614,15 @@ void runTransaction() {
     node_4.dirtyFlags = 0;
     node_5.dirtyFlags = 0;
     node_6.dirtyFlags = 0;
+
+    // Сlean errors from pulse outputs
+    if (node_6.outputHasError_DONE) {
+      node_6.outputHasError_DONE = false;
+#if defined(XOD_DEBUG) || defined(XOD_SIMULATION)
+      detail::printErrorToDebugSerial(6, node_6.errorFlags);
+#endif
```

Comment:
These `#if ... printError #endif` are all over the source. They ugly a bit. What if you’ll guard the *body* of  `detail::printErrorToDebugSerial` and omit the checks at all use points?

If we’re not in simulation, it would be an empty function call which, I’m pretty sure, will be thrown away by C++ optimizer.

A bulletproof solution is creating a macro like `XOD_TRACE`, but for error purposes. For example, `XOD_DEBUG_ERROR(6, node_6.errorFlags)`. It can be expanded to no-op if we’re not simulating.

Reply:
Moved `#if ... printError #endif` into the body of `detail::printErrorToDebugSerial`.